pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -196,8 +202,13 @@ func removeSelectorSpaces(selectorName string) string {
 }
 
 func (c *DeviceController) refreshMediatedDevicesTypes() {
-	nodeDesiredMdevTypesList := c.virtConfig.GetDesiredMDEVTypes(c.host)
-	err := c.mdevTypesManager.updateMDEVTypesConfiguration(nodeDesiredMdevTypesList)
+	node, err := c.clientset.Nodes().Get(context.Background(), c.host, metav1.GetOptions{})
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to configure the desired mdev types, failed to get node details")
+		return
+	}
```

Comment:
it's possible we might need to consider being more persistent here if the API command fails to get the node.

From what I understand, this `refreshMediatedDevicesType` get's kicked off when the KV config map changes. So that's a one shot deal when something is updated. If we fail here, it won't be retried.

Also, what happens if the node is updated with new labels and that makes it match different mediated devices? I don't think there's anything triggering an update when the node changes. Maybe this logic should be tied to the node heartbeat or something, where we're already accessing the node object on a recurring interval.

Reply:
Yep, thanks! This makes sense to me now.